openstudiocore/src/energyplus/ForwardTranslator/ForwardTranslateCoilHeatingDesuperheater.cpp

```diff
@@ -104,13 +101,36 @@ boost::optional<IdfObject> ForwardTranslator::translateCoilHeatingDesuperheater(
   boost::optional<ModelObject> heatingSource = modelObject.heatingSource();
 
   if ( heatingSource ) {
-      boost::optional<IdfObject> _heatingSource = translateAndMapModelObject(heatingSource.get());
+    if (boost::optional<IdfObject> _heatingSource = translateAndMapModelObject(*heatingSource)) {
+
+      // Note JM 2019-03-14:
+      // If the coil in question is a DX coil (CoilCoolingDXSingleSpeed, CoilCoolingDXTwoSpeed, CoilCoolingDXTwoStageWithHumidityControlMode)
+      // and this DX coil isn't already wrapped in a Unitary, then the FT will wrap it into a CoilSystem:Cooling:DX object and return that, but we
+      // need the DX coil here and not the wrapper.
+      //
+      // Note: Other accepted types are Refrigeration objects and don't suffer the same problem
+      // (Refrigeration:Condenser:AirCooled, Refrigeration:Condenser:EvaporativeCooled, Refrigeration:Condenser:WaterCooled)
+
+      std::string objectType;
+      std::string objectName;
+
+      if (_heatingSource->iddObject().type() == IddObjectType::CoilSystem_Cooling_DX) {
+        // We have retrieve the coil itself, not the wrapper
+        objectType = _heatingSource->getString(CoilSystem_Cooling_DXFields::CoolingCoilObjectType).get();
+        objectName = _heatingSource->getString(CoilSystem_Cooling_DXFields::CoolingCoilName).get();
+      } else {
+        objectType = _heatingSource->iddObject().name();
+        objectName = _heatingSource->name().get();
+      }
 
-      if ( _heatingSource && _heatingSource->name() )
-      {
-        object.setString(Coil_Heating_DesuperheaterFields::HeatingSourceObjectType,_heatingSource->iddObject().name());
-        object.setString(Coil_Heating_DesuperheaterFields::HeatingSourceName,_heatingSource->name().get());
+      if (objectType.empty() || objectName.empty()) {
+        LOG(Error, "Something went wrong in the translation of " << modelObject.briefDescription() << ", couldn't retrieve the coil");
```

Comment:
Why not issue this log message, but assign the empty strings to the new object and let it fail at E+ run time?

Reply:
Isn't it already what will happen unless you have a debug build? I don't assign the string in question so they come up empty, and if in release you don't hit the OS_ASSERT, so E+ will end up failing.

Reply:
Commented out in a03e5f42b71df2536194af85578f5f95d262285e